src/components/NotFound.astro

```diff
@@ -156,3 +156,15 @@ const dictionary = Object.fromEntries(
 	}
 	customElements.define('four-oh-four', FourOhFour);
 </script>
+
+<script is:inline>
+	// Script is inlined to avoid visible flash
+	(() => {
+		// Get current language from URL, then apply it to language selector if matched
+    const cssSelector = `starlight-lang-select option[value="${location.pathname}"]`;
```

Comment:
I don’t think this actually works in production. See https://docs-git-fork-mingjunlu-fix-language-selec-3866c6-astrodotbuild.vercel.app/de/x/ for example where the language is German, but the language select remains on English.

This is because the page is served statically and the language select always moves between routes like `/en/404/` and `/de/404/`, not the current page’s location. Instead we should probably parse the language segment and find the option that matches `/${lang}/404/`.

(In dev mode your approach works because the page is rendered on-demand and uses the actual URL instead of the `/404/` one.)

Reply:
Thanks for the explanation! I've made some [adjustments](https://github.com/withastro/docs/pull/6763/commits/d03a4b97431b66e83ed09be8d5f39e25f4f1146d#diff-3f71d04203961deb2eef1334ac685c4596b67a98a96f0551b33fcf2a454cbd91R164-R165).

Instead of looking for an exact match (`[value="/${lang}/404/"]`), I changed the CSS selector to match the beginning part only (`[value^="/${lang}/"]`), so that things would work in both production and development.